### PR TITLE
Header

### PR DESCRIPTION
**Pull request by FilipMasar**

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -15,6 +12,15 @@ const useStyles = createStyles(theme => ({
   },
 }))
 
+const config: Config = {
+  readOnlyChainId: Mumbai.chainId,
+  readOnlyUrls: {
+    [Mumbai.chainId]: process.env.RPC_URL_80001 ?? '',
+    [Localhost.chainId]: process.env.RPC_URL_1337 ?? '',
```
Comment (mondejka) [inline pages/_app.tsx:19]: Add localhost only in dev environment

File: components/header/web3Connect/index.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+  }, [error])
+
+  const activate = async () => {
+    setActivateError(undefined)
+    activateBrowserWallet()
+  }
+
+  return (
+    <div className={classes.root}>
```
Comment (mondejka) [inline components/header/web3Connect/index.tsx:43]: You can use [group](https://mantine.dev/core/group/), so no css will be needed.

File: components/header/web3Connect/Account.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+const Account = () => {
+  const { classes } = useStyles()
+  const { account, chainId } = useEthers()
+  const config = useConfig()
+  const ens = useLookupAddress()
+  const [showModal, setShowModal] = useState(false)
+  const etherBalance = useEtherBalance(account)
+
+
+  if (chainId && config.networks && config.networks.every((chain) => chain.chainId !== chainId)) {
```
Comment (mondejka) [inline components/header/web3Connect/Account.tsx:28]:
```suggestion
  if (chainId && !config?.networks.includes(chainId)) {
```

File: components/header/web3Connect/Account.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    lineHeight: 0,
+  }
+}))
+
+const Account = () => {
+  const { classes } = useStyles()
+  const { account, chainId } = useEthers()
+  const config = useConfig()
+  const ens = useLookupAddress()
+  const [showModal, setShowModal] = useState(false)
```
Comment (mondejka) [inline components/header/web3Connect/Account.tsx:24]:
```suggestion
  const [showModal, modalHandler] = useDisclosure(false)
```

**Review by mondejka — APPROVED** [review]
nitpicks

File: components/header/web3Connect/Account.tsx (hunk 3 above)
Comment (FilipMasar) [reply components/header/web3Connect/Account.tsx:28]: `chainId` is number, while `config.networks` contains array of `Chain` objets

Comment (mondejka) [reply components/header/web3Connect/Account.tsx:28]: Ah sorry didn't notice, but still this is more readable IMHO.
```suggestion
!config?.networks.map(({chainId}) => chainId).includes(chainId)
```

File: utils/DappProviderConfig.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  readOnlyChainId: Mumbai.chainId,
+  readOnlyUrls: {
+    [Mumbai.chainId]: process.env.RPC_URL_80001 ?? '',
+    ...(dev && { [Localhost.chainId]: process.env.RPC_URL_1337 ?? '' }),
+  },
+  networks: [
+    Mumbai,
+    ...(dev ? [Localhost] : []),
+  ]
+}
```
Comment (mondejka) [inline utils/DappProviderConfig.ts:15]:
```javascript
networks: [Mumbai, dev && Localhost].filter(identity),
```

**Review by mondejka — APPROVED** [review]

File: utils/DappProviderConfig.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { Config, Localhost, Mumbai } from "@usedapp/core"
+
+const dev = process.env.NODE_ENV === "development"
+
+const config: Config = {
+  readOnlyChainId: Mumbai.chainId,
+  readOnlyUrls: {
+    [Mumbai.chainId]: process.env.RPC_URL_80001 ?? '',
+    ...(dev && { [Localhost.chainId]: process.env.RPC_URL_1337 ?? '' }),
+  },
```
[the remark this answers is not in the document]
Comment (FilipMasar) [reply utils/DappProviderConfig.ts]: I struggled using lodash. Pls check my solution

**Pull request merged**